src/apollos/models/users/model.js

```diff
@@ -46,6 +54,229 @@ export class User {
+
+  async loginUser(user = {}) {
+    try {
+      if (!user.IsConfirmed) {
+        api.post(`/UserLogins/${user.Id}`, {
+          data: {
+            IsConfirmed: true,
+          },
+        });
+      }
+
+      api.patch(`/UserLogins/${user.Id}`, {
+        data: {
+          LastLoginDateTime: `${moment().toISOString()}`,
+        },
+      });
+
+      return user;
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  createUserProfile(props = {}) {
+    const {
+      email,
+      firstName,
+      lastName,
+    } = props;
+
+    return api.post("/People", {
+      data: {
+        Email: email,
+        Guid: makeNewGuid(),
+        FirstName: stripTags(firstName),
+        LastName: stripTags(lastName),
+        IsSystem: false,
+        Gender: 0,
+        RecordTypeValueId: 1,
+        ConnectionStatusValueId: 67, // Web Prospect
+        SystemNote: "Created from NewSpring Apollos",
+      },
+    });
+  }
+
+  createUser(props = {}) {
+    const {
+      email,
+      password,
+      personId,
+    } = props;
+
+    return api.post("/UserLogins", {
+      data: {
+        PersonId: personId,
+        EntityTypeId: 27,
+        UserName: email,
+        IsConfirmed: true,
+        PlainTextPassword: password,
+        LastLoginDateTime: `${moment().toISOString()}`,
+      },
+    });
+  }
+
+  async registerUser(props = {}) {
+    try {
+      const {
+        email,
+        firstName,
+        lastName,
+        password,
+      } = props;
+
+      const personId = await this.createUserProfile({
+        email,
+        firstName,
+        lastName,
+      });
+
+      const userId = await this.createUser({
+        email,
+        password,
+        personId,
+      });
+
+      const [user, person] = await Promise.all([
+        api.get(`/UserLogins/${userId}`),
+        api.get(`/People/${personId}`),
+      ]);
+
+      // Removed saving it to a constant
+      // because we don't know if this ID could change
+      // from other sources (migrations, integrity checks, etc)
+      const [systemEmail] = await api.get("/SystemEmails?$filter=Title eq 'Account Created'");
+      if (!systemEmail) throw new Error("Account created email does not exist!");
+
+      await sendEmail(
+        systemEmail.Id,
+        Number(person.PrimaryAliasId),
+        {
+          Person: person,
+          User: user,
+        },
+      );
+
+      return userId;
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  async forgotPassword(username, sourceURL) {
+    try {
+      const user = await this.getByUsername(username);
+      if (!user) throw new Error("User does not exist!");
+
+      const person = await api.get(`People/${user.PersonId}`);
+      if (!person) throw new Error("User profile does not exist!");
+
+      const token = Random.secret();
+
+      // Not sure if I can do this (also can't expire this way until after use)
+      await api.put(`UserLogins/${user.Id}`, {
+        ...user,
+        ResetPasswordToken: token,
```

Comment:
Not sure if I can add just any arbitrary key. If this touches MSSQL and the column doesn't exist it shouldn't add the column unless the endpoint has some magic going on that'll handle this. If we can't do this, where can we store the users' token?